testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -866,6 +878,181 @@ protected void checkInformationSchemaViewsForMaterializedView(String schemaName,
                 .containsAll("VALUES '" + viewName + "'");
     }
 
+    /**
+     * Test that reading table, column metadata, like {@code SHOW TABLES} or reading from {@code information_schema.views}
+     * does not fail when relations are concurrently created or dropped.
+     */
+    @Test(timeOut = 180_000)
+    public void testReadMetadataWithRelationsConcurrentModifications()
+            throws Exception
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE) && !hasBehavior(SUPPORTS_CREATE_VIEW) && !hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            throw new SkipException("Cannot test");
+        }
+
+        int readIterations = 10;
+        // generous timeout as this is a generic test; typically should be faster
+        int testTimeoutSeconds = 150;
```

Comment:
This isn't timeout for the entire test - it's just for the test setup (i.e. first create + drop starts running). Can you rename to clarify?

Reply:
No, it's used after setup too, see
```
completionService.poll(testTimeoutSeconds - testWatch.elapsed(SECONDS), SECONDS)
```

Reply:
Good point. Upper bound for test (at-least the tasks) is decided by this value.